### ISSUE TITLE
Viem support?

### ISSUE THREAD

**Issue by mlenser** [spruceid/siwe#173]
Is there any plan to support [Viem](https://viem.sh/docs/introduction.html)? It's made by the `wagmi` team as an alternative to ethers.

**Comment by magiclars-off** [comment]
+1

**Comment by conwayconstar** [comment]
I created a view version here https://github.com/feelsgoodman-web3/siwveim

**Comment by w4ll3** [comment]
Hello, we are discussing how to address this, since the library went through an audit it gets a little more complicated to replace such a vital library. Hope you understand!

**Comment by reslear** [comment]
in the end, we should 👍

**Comment by FrancoAguzzi** [comment]
+1 vote to have viem support 🙏🏼

**Comment by reslear** [comment]
also found https://github.com/softwarecurator/eip-login by @softwarecurator
but it's probably worth rewriting to viem 2.

**Comment by reslear** [comment]
So i write small lib, also welcome to testing simplified siwe with **Viem 2** 

https://github.com/reslear/simple-siwe

**Comment by 2facedsquid** [comment]
any update on this?

**Comment by sbihel** [comment]
We are not planning to add support for viem but would welcome a contribution. As this library already supports both ethers.js v5 and v6 I believe the "crypto backend" is abstracted enough to make such addition relatively easily, unless I'm missing something.

**Comment by conwayconstar** [comment]
Say no more!

**Comment by reslear** [comment]
@nfts2me Yes and no, there's also a deps problem (like parsers), See my Comparison above packages on viem ~2mb. 

however siwe 2.8 + ethers 18.8MB = ~20mb

hence the motivation to make https://github.com/reslear/simple-siwe a scaled-down copy without dependencies

**Comment by reslear** [comment]
Good news https://github.com/wevm/viem/pull/2280

**Comment by reslear** [comment]
we can close the subject @sbihel
